mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/MobileAcademyServiceImpl.java

```diff
@@ -457,4 +470,60 @@ public String getScoresForUser(Long callingNumber) {
 
         return scores;
     }
-}
+
+    @Override
+    public void updateMsisdn(Long oldCallingNumber, Long newCallingNumber) {
+
+        if (oldCallingNumber.equals(newCallingNumber)) {
+            return;
+        }
+        // Update Msisdn  In MTRAINING_MODULE_BOOKMARK
+        LOGGER.debug("Fetching Bookmarks for Msisdn {}.", oldCallingNumber);
+        List<Bookmark> existingBookmarks = bookmarkService.getAllBookmarksForUser(oldCallingNumber.toString());
+        if (existingBookmarks.size() > 0) {
+            int i;
+            Bookmark bookmark;
+            for (i = 0; i < existingBookmarks.size(); i++) {
+                bookmark = existingBookmarks.get(i);
+                bookmark.setExternalId(newCallingNumber.toString());
+                bookmarkService.updateBookmark(bookmark);
+            }
+            LOGGER.debug("Updated MSISDN {} to {} in {} Bookmarks", oldCallingNumber, newCallingNumber, i);
+        } else {
```

Comment:
We can have the ideal case once the database is fixed (NMS-369). But we can't wait till the issue is fixed as Rajasthan State Govt is going to release BSNL SIM cards for all the FLWs in the state. Thus we need to deploy the code changes immediately. We do not expect these numbers to overlap with existing MSISDNs, thus BBCMA has been insisting that we deploy the code changes ASAP.